### PR TITLE
Fix OnItemChecked method throw exception when the ListViewItem is clone object.

### LINKED ISSUE #11658: InvalidOperationException thrown when calling OnItemChecked in class derived from ListView (in VirtualMode)
### .NET version

.NET 8

### Did it work in .NET Framework?

Yes

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

_No response_

### Issue description

We have a class derived from ListView. It calls OnItemChecked in order to raise the ItemChecked event to subscribers. In the ItemCheckedEventArgs we pass in a cloned ListViewItem so that we can set the Checked property correctly. This worked fine in .NET Framework as OnItemChecked just raised the event. However, in .NET 8, it's throwing an InvalidOperationException:
"Cannot get accessiblity object when ListViewItem is not attached to a ListView."
An accessibility object has been created for the ListView (which seems reasonable), however, the cloned ListViewItem is not hosted by the ListView, which I think is the cause of the exception.

### Steps to reproduce

Derive a class from ListView
Ensure VirtualMode and Checkboxes are enabled
Ensure the accessibility object is created (not sure how to force this?)
Add a method that calls OnItemChecked - something like:

```
ListViewItem lvi = HitTest(...); // Any code that gets a ListViewItem from the ListView
var update = (ListViewItem) lvi.Clone();
update.Checked = true;
Debug.Assert(update.Checked);
Debug.Assert(update.ListView == null);
OnItemChecked(new ItemCheckedEventArgs(update)); // throws exception
```

### PR DESCRIPTION
**Pull request by SimonZhao888**
Fixes #11658

Add `e.item.ListView == this` to the OnItemChecked method to avoid throwing an accessiblity object not created exception when the ListViewItem is a clone.

## Proposed changes

- Verify that this item belongs to this ListView in OnItemChecked method.

## Customer Impact

- When the ListViewItem is a clone, the OnItemChecked method does not throw an accessiblity object not created exception.

## Regression? 

- Yes

## Risk

- Minimal

## Test methodology 

- Manually

## Test environment(s) 

- 9.0.100-preview.7.24407.12

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11892)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/Controls/ListView/ListView.cs [hunk 1]
```diff
@@ -4835,7 +4835,7 @@ protected virtual void OnItemChecked(ItemCheckedEventArgs e)
             return;
         }
 
-        if (IsAccessibilityObjectCreated)
+        if (IsAccessibilityObjectCreated && e.Item.ListView == this)
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/Controls/ListView/ListView.cs:4838]:
```suggestion
        if (e.Item.ListView == this && IsAccessibilityObjectCreated)
```

I think this test `e.Item.ListView == this` is faster, so the condition will complete faster if it fails

**Review by Tanya-Solyanik — COMMENTED** [review]
Looks good. @lonitra  - I suggest taking it to RC2, what do you think?

**Comment by lonitra** [comment]
> I suggest taking it to RC2, what do you think?

This doesn't seem to be a regression, but we can try to take to RC2 this looks accessibility related, there doesn't seem to be a workaround, and customer has hit this scenario. Mail needs to be sent to tactics for this to be merged into release/9.0

**Comment by Tanya-Solyanik** [comment]
It's a regression from .NET framework

**Review by Tanya-Solyanik — APPROVED** [review]

**Pull request merged**